pkg/qliksense/context_configs.go

```diff
@@ -422,3 +428,80 @@ func validateInput(input string) (string, error) {
 	}
 	return input, err
 }
+
+// PrepareK8sSecret decodes and decrypts the secret value in the secret.yaml file and returns a B64encoded string
+func (q *Qliksense) PrepareK8sSecret(qliksenseCR api.QliksenseCR, targetFile string) (string, error) {
+	secretKeyPairLocation := q.GetSecretKeyPairLocation(qliksenseCR)
+	privateKeyFile := filepath.Join(secretKeyPairLocation, api.QliksensePrivateKey)
+
+	// check if private key file  and targetFile exists
+	if !api.FileExists(privateKeyFile) || !api.FileExists(targetFile) {
+		err := fmt.Errorf("Either private key file or target file does not exist in the path provided")
+		log.Println(err)
+		return "", err
+	}
+
+	// read the target file and private key
+	k8sSecret, privateKeybytes, err := readPrivateKeyAndTargetfile(privateKeyFile, targetFile)
+	if err != nil {
+		return "", err
+	}
+	// retrieve value from data section
+	k8sSecret1, err := api.K8sSecretFromYaml(k8sSecret)
+	if err != nil {
+		return "", err
+	}
+
+	// convert []byte into RSA public key object
+	rsaPrivateKey, err := api.DecodeToPrivateKey(privateKeybytes)
+	if err != nil {
+		return "", err
+	}
+	dataMap := k8sSecret1.Data
+	var base64EncodedSecret string
+	var resultMap = make(map[string]string)
+	for k, v := range dataMap {
+		// base64 decode every value
+		decodedStr, _ := base64.StdEncoding.DecodeString(string(v))
+
+		decryptedString, err := api.Decrypt(decodedStr, rsaPrivateKey)
+		if err != nil {
+			err := fmt.Errorf("Not able to decrypt message")
+			return "", err
+		}
+
+		// base64 encode the values
+		base64EncodedSecret = b64.StdEncoding.EncodeToString(decryptedString)
+		// resultMap[k] = ([]byte)base64EncodedSecret
+		resultMap[k] = base64EncodedSecret
+
+	}
+	api.LogDebugMessage("B64 encoded Map: %v\n", resultMap)
+
+	// putting the above map back into the k8sSecret struct
+	k8sSecret1.StringData = resultMap
```

Comment:
isnt it `k8sSecret1.Data` instead of `k8sSecret1.StringData`

Reply:
fixed